### PR TITLE
location aliasing

### PR DESCRIPTION
**Pull request by jimihem**
when location aliasing, the aliases sharing the location must have the same underlying numerical type and bit width (floating-point or integer, 32-bit versus 64-bit, etc.) and the same auxiliary storage and interpolation qualification.

### CODE REVIEW

**Comment by jimihem** [comment]
This issue has been fixed once before, but the consideration was not comprehensive enough
#3438

**Comment by jimihem** [comment]
@pmistryNV
@svenvh

File: Test/ps_sample.frag [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+out vec4 fs_out;
+
+void main()
+{
+    vec4 result = gs_fs;
+
+
+
+    fs_out = result;
+}
```
Comment (svenvh) [inline Test/ps_sample.frag:17]: Missing newline at EOF.

File: glslang/MachineIndependent/linkValidate.cpp [hunk 2]
```diff
@@ -1736,7 +1736,19 @@ int TIntermediate::addUsedLocation(const TQualifier& qualifier, const TType& typ
...
 }
-
+static bool checkType(TBasicType t1, TBasicType t2) {
+    if (t1 != t2) {
+        if ((t1 == EbtInt8 && t2 == EbtUint8) ||
+            (t2 == EbtInt8 && t1 == EbtUint8) ||
+            (t1 == EbtInt16 && t2 == EbtUint16) ||
+            (t2 == EbtInt16 && t1 == EbtUint16)||
+            (t1 == EbtInt && t2 == EbtUint) ||
+            (t2 == EbtInt && t1 == EbtUint)) {
```
Comment (svenvh) [inline glslang/MachineIndependent/linkValidate.cpp:1746]: You should also add cases for Ebt(U)Int64.

**Review by svenvh — COMMENTED** [review]
Thanks for the followup PR!  Just a few minor improvements.

**Review by svenvh — APPROVED** [review]
Thanks, LGTM.  The latest patch solves the issue I reported in #3438

**Review by arcady-lunarg — APPROVED** [review]
LGTM, thanks for adding the tests.

**Pull request merged**